### PR TITLE
[Blocks] Migrate Price filter tests to Playwright

### PR DESCRIPTION
**Pull request by gigitux**
> [!WARNING]  
> This PR is blocked by #46125

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR migrates the Price Filter tests to Plauywright infrastructure.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that E2E tests are green.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 
Migrate Price filter tests to Playwright

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+		);
+
+		const allProducts = allProductsBlock.getByRole( 'listitem' );
+
+		await expect( allProducts ).toHaveCount( 1 );
+		expect( page.url() ).toContain(
+			blockData.urlSearchParamWhenFilterIsApplied
+		);
+	} );
+} );
```
Comment (gigitux) [inline plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts]: This code was already in place

File: plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+		await editor.insertBlock( {
+			name: 'woocommerce/filter-wrapper',
+			attributes: {
+				filterType: 'price-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await editor.publishPost();
+		const url = new URL( page.url() );
+		const postId = url.searchParams.get( 'post' );
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts]: `editor.publishPost()` returns the post ID so we can make a oneliner here:
```suggestion
		const postId = await editor.publishPost();
```

File: plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+			.waitFor( { timeout: 3000 } )
+			.catch( () => {
+				// Do not throw in case Playwright doesn't make it in time for the
+				// initial (pre-request) render.
+			} );
+
+		const maxPriceInput = page.getByRole( 'textbox', {
+			name: 'Filter products by maximum price',
+		} );
+		// await page.pause();
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts]: 🗑️

File: plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+			'npm run wp-env run tests-cli -- wp option update wc_blocks_use_blockified_product_grid_block_as_template false'
+		);
+
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//archive-product',
+			postType: 'wp_template',
+			canvas: 'edit',
+		} );
+
+		await editor.canvas.locator( 'body' ).click();
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts]: We're already loading with `canvas: edit` so we should be good removing this line.

**Review by WunderBart — CHANGES_REQUESTED** [review]
Just a few comments!

**Review by WunderBart — APPROVED** [review]

**Pull request merged**